dts/bindings/pwm/nxp,imx-pwm.yaml

```diff
@@ -33,6 +33,17 @@ properties:
       description: |
         Clock prescaler at the input of the PWM.
 
+    nxp,reload:
+      type: string
+      # enum order should be accord HAL enum pwm_register_reload_t
+      enum:
+        - "Immediate"
+        - "HalfCycle"
+        - "FullCycle"
+        - "HalfAndFullCycle"
+      description: |
+        Select how to load the buffered-registers with new values.
```

Comment:
Not sure I like the binding having to match the enum values in the NXP HAL. This seems very fragile. Also, we tend to use lower-case for enums in DTS - and the description needs to explain what the different settings mean.

Reply:
This is matching the hardware register options as well. I feel these options should be fine. We should change to use lower-case.

Reply:
@GrixaYrev can you help update

Reply:
@mmahadevan108, please check:
```
    nxp,reload:
      type: string
      enum:
        - "immediate"
        - "half-cycle"
        - "full-cycle"
        - "half-and-full-cycle"
      description: |
        Select how to load the buffered-registers with new values:
        "immediate" - buffered-registers get loaded with new values as soon as LDOK bit is set;
        "half-cycle" - registers loaded on a PWM half cycle;
        "full-cycle" - registers loaded on a PWM full cycle;
        "half-and-full-cycle" - registers loaded on a PWM half & full cycle.
```
I agree with @henrikbrixandersen and i will add convertation of this binding's enum to NXP HAL enum. It will be more robust.

Reply:
Looks good to me, can you help update this PR?

Reply:
@mmahadevan108, i has waited your approve and now i ready to update PR.